src/controllers/FacetQueryController.ts

```diff
@@ -270,18 +270,25 @@ export class FacetQueryController {
 
     if (this.queryOverrideIsNeededForMultiSelection()) {
       queryBuilderExpression = this.processQueryOverrideForMultiSelection(queryBuilder, queryBuilderExpression);
+    } else {
+      queryBuilderExpression.reset();
     }
     if (this.queryOverrideIsNeededForAdditionalFilter()) {
       queryBuilderExpression = this.processQueryOverrideForAdditionalFilter(queryBuilder, queryBuilderExpression);
     }
 
     queryBuilderExpression = this.processQueryOverrideForEmptyValues(queryBuilder, queryBuilderExpression);
-
     return queryBuilderExpression;
   }
 
   private queryOverrideIsNeededForMultiSelection() {
-    return !(this.facet.options.useAnd || (this.facet.options.isMultiValueField && this.facet.values.hasSelectedAndExcludedValues()));
+    if (this.facet.options.useAnd) {
+      return false;
+    }
+    if (this.facet.values.hasSelectedOrExcludedValues()) {
```

Comment:
I am not sure if this is part of the fix, but if we want to keep the same logic as the original source code, then this `if` condition should be:
```
if (this.facet.options.isMultiValueField && this.facet.values.hasSelectedAndExcludedValues()) {
    return false;
}
```

and the final statement should be `return true`.

Reply:
Original was not correct